tests/Dfc.CourseDirectory.WebV2.Tests/FeatureTests/NewApprenticeshipProvider/HidePassedNotificationTests.cs

```diff
@@ -0,0 +1,232 @@
+﻿using Dfc.CourseDirectory.WebV2.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Dfc.CourseDirectory.WebV2.Tests.FeatureTests.NewApprenticeshipProvider
+{
+    public class HidePassedNotificationTests : TestBase
+    {
+        public HidePassedNotificationTests(CourseDirectoryApplicationFactory factory)
+            : base(factory)
+        {
+        }
+
+        [Fact]
+        public async Task Post_InvalidSubmissionReturnsBadRequest()
+        {
+            // Arrange
+            var providerId = await TestData.CreateProvider(apprenticeshipQAStatus: ApprenticeshipQAStatus.Passed);
+            await User.AsHelpdesk();
+
+            // Act
+            var response = await HttpClient.PostAsync(
+                $"new-apprenticeship-provider/hide-passed-notification?providerId={providerId}&returnUrl=/",null);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+        }
+
+        [Theory]
+        [InlineData(ApprenticeshipQAStatus.Submitted)]
+        [InlineData(ApprenticeshipQAStatus.InProgress)]
+        [InlineData(ApprenticeshipQAStatus.Failed)]
+        [InlineData(ApprenticeshipQAStatus.UnableToComplete)]
+        public async Task Post_NonePassedQAStatusReturnsBadRequest(ApprenticeshipQAStatus status)
+        {
+            // Arrange
+            var ukprn = 12345;
+            var email = "somebody@provider1.com";
+            var providerName = "Provider 1";
+            var providerUserId = $"{ukprn}-user";
+            var adminUserId = $"admin-user";
+            Clock.UtcNow = new DateTime(2019, 5, 17, 9, 3, 27, DateTimeKind.Utc);
+            var requestedOn = Clock.UtcNow;
+
+            var providerId = await TestData.CreateProvider(
+                ukprn: ukprn,
+                providerName: providerName,
+                apprenticeshipQAStatus: status);
+
+            await TestData.CreateUser(providerUserId, email, "Provider 1", "Person", providerId);
+            await TestData.CreateUser(adminUserId, "admin", "admin", "admin", null);
+            var apprenticeshipId = await TestData.CreateApprenticeship(ukprn);
+
+            //create submission
+            var submissionId = await TestData.CreateApprenticeshipQASubmission(
+                providerId,
+                submittedOn: Clock.UtcNow,
+                submittedByUserId: providerUserId,
+                providerMarketingInformation: "The overview",
+                apprenticeshipIds: new[] { apprenticeshipId });
+
+            //update qa submission
```

Comment:
Need to make sure the created data here is valid w.r.t to the QA status:
if Submitted then there shouldn't be any assessments;
if InProgress then there should be at least one assessment (provider or assessment) and submission `Passed` should be null;
if Failed then there should be both assessments and at least one should be Failed and the submission itself should have `Passed` == false;
if UnableToComplete - this is status isn't valid by itself - it should be combined with another status e.g. `ApprenticeshipQAStatus.UnableToComplete | ApprenticeshipQAStatus.NotStarted` then the rules for the combined status apply.
Need to add a check for status == NotStarted. There should be no submission in this case.

Reply:
done